### ISSUE TITLE
cannot find required auxiliary files: compile

### ISSUE THREAD

**Issue by callimac-o** [JKBert/knoda5-hkc#1]
Ciao,
in a new debian11 i'm triyng to install hk_classes but an error occurred:

callimaco@deb-portatile:~/Documenti/hk_classes/knoda5-hkc-main$ ./configure
configure: error: cannot find required auxiliary files: compile

can you help me?

ciao grazie Callimaco

**Comment by JKBert** [comment]
callimac-o a écrit :
> Ciao,
> in a new debian11 i'm triyng to install hk_classes but an error occurred:
>
> ***@***.***:~/Documenti/hk_classes/knoda5-hkc-main$ ./configure
> configure: error: cannot find required auxiliary files: compile
>
> can you help me?
>
> ciao grazie Callimaco

	Hello,

	You should run automake --add-missing and autoreconf. Generated scripts
are created on my workstation with my version of autotools.

	Regards,

	JKB

**Comment by callimac-o** [comment]
Ciao,
merci beaucoup for your fast reply.
I'm not expert with software compilation, can you explai me what i have to do?
grazie Callimaco

**Comment by JKBert** [comment]
callimac-o a écrit :
> Ciao,
> merci beaucoup for your fast reply.
> I'm not expert with software compilation, can you explai me what i have
> to do?

	Some files depend on your automake release. Thus you have to launch
automake -f --add-missing to replace and add some tools and process .am
files into .in. After that, you can run autoreconf to generate configrue
script from configure.in or configure.ac.

	Regards,

	JKB

**Comment by callimac-o** [comment]
Merci,

i'll try and i'll know you

Grazie Callimaco

**Comment by callimac-o** [comment]
ciao

by running automake -f --add-missing only and not autoconf the ./configure command works. Unfortunately, however, automake does not work which gives me this error:

xbase / xblock.cpp: In member function 'virtual xbShort foxproLock :: LockIndex (xbShort)':
xbase / xblock.cpp: 563: 40: warning: unused parameter 'LockType' [-Wunused-parameter]
   563 | xbShort foxproLock :: LockIndex (xbShort LockType)
       | ~~~~~~~~ ^ ~~~~~~~
make [2]: *** [Makefile: 647: xblock.lo] Error 1
make [2]: exit from directory "/ home / callimaco / Documents / hk_classes / knoda5-hkc-main / hk_dbaseclasses"
make [1]: *** [Makefile: 445: all-recursive] Error 1
make [1]: exit directory "/ home / callimaco / Documents / hk_classes / knoda5-hkc-main"
make: *** [Makefile: 377: all] Error 2
callimaco @ deb-portable: ~ / Documents / hk_classes / knoda5-hkc-main $ 

grazie ciao Callimaco

**Comment by JKBert** [comment]
callimac-o a écrit :
> ciao
>
> by running automake -f --add-missing only and not autoconf the
> ./configure command works. Unfortunately, however, automake does not
> work which gives me this error:
>
> xbase / xblock.cpp: In member function 'virtual xbShort foxproLock ::
> LockIndex (xbShort)':
> xbase / xblock.cpp: 563: 40: warning: unused parameter 'LockType'
> [-Wunused-parameter]
> 563 | xbShort foxproLock :: LockIndex (xbShort LockType)
> | ~~~~~~~~ ^ ~~~~~~~
> make [2]: *** [Makefile: 647: xblock.lo] Error 1
> make [2]: exit from directory "/ home / callimaco / Documents /
> hk_classes / knoda5-hkc-main / hk_dbaseclasses"
> make [1]: *** [Makefile: 445: all-recursive] Error 1
> make [1]: exit directory "/ home / callimaco / Documents / hk_classes /
> knoda5-hkc-main"
> make: *** [Makefile: 377: all] Error 2
> callimaco @ deb-portable: ~ / Documents / hk_classes / knoda5-hkc-main $
>

	Some errors remain in xbase and maybe other drivers. Try to configure
with --without-xbase. If I remember, mysql and postgresql drivers run as
expected but there are some issues with other drivers (compilation
error). I have to convert code into modern C++ dialect.

	JKB

**Comment by callimac-o** [comment]
I tried installing just sqlite3 (which is the db I use); in this case it gives me this error

make [2]: exit from directory "/ home / callimaco / Documents / hk_classes / knoda5-hkc-main / hk_sqlite3classes"
Making all in utilities
make [2]: entry into directory "/ home / callimaco / Documents / hk_classes / knoda5-hkc-main / utilities"
/ bin / bash ../libtool --tag = CXX --mode = link g ++ -Wall -W -Wconversion -Wshadow -Wcast-qual -Wwrite-strings -I / usr / include / libxml2 -m64 -g -O2 - o hk_actionquery hk_actionqueryutility.o ../hk_classes/libhk_classes.la -ldl
libtool: link: g ++ -Wall -W -Wconversion -Wshadow -Wcast-qual -Wwrite-strings -I / usr / include / libxml2 -m64 -g -O2 -o .libs / hk_actionquery hk_actionqueryutility.o ../hk_classes/. libs / libhk_classes.so -ldl -Wl, -rpath -Wl, / usr / lib64 / hk_classes
/ usr / bin / ld: hk_actionqueryutility.o: undefined reference to symbol 'xmlParseFile @@ LIBXML2_2.4.30'
/ usr / bin / ld: /lib/x86_64-linux-gnu/libxml2.so.2: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make [2]: *** [Makefile: 514: hk_actionquery] Error 1
make [2]: exit directory "/ home / callimaco / Documents / hk_classes / knoda5-hkc-main / utilities"
make [1]: *** [Makefile: 445: all-recursive] Error 1
make [1]: exit from directory "/ home / callimaco / Documents / hk_classes / knoda5-hkc-main"
make: *** [Makefile: 377: all] Error 2

**Comment by callimac-o** [comment]
sorry! is this the error:
Making all in utilities
make[2]: ingresso nella directory «/home/callimaco/Documenti/hk_classes/knoda5-hkc-main/utilities»
/bin/bash ../libtool  --tag=CXX   --mode=link g++ -Wall   -W -Wconversion -Wshadow -Wcast-qual -Wwrite-strings -I/usr/include/libxml2 -m64 -g -O2   -o hk_actionquery hk_actionqueryutility.o ../hk_classes/libhk_classes.la -ldl 
libtool: link: g++ -Wall -W -Wconversion -Wshadow -Wcast-qual -Wwrite-strings -I/usr/include/libxml2 -m64 -g -O2 -o .libs/hk_actionquery hk_actionqueryutility.o  ../hk_classes/.libs/libhk_classes.so -ldl -Wl,-rpath -Wl,/usr/lib64/hk_classes
/usr/bin/ld: hk_actionqueryutility.o: undefined reference to symbol 'xmlParseFile@@LIBXML2_2.4.30'
/usr/bin/ld: /lib/x86_64-linux-gnu/libxml2.so.2: error adding symbols: DSO missing from command line
collect2: error: ld returned 1 exit status
make[2]: *** [Makefile:514: hk_actionquery] Errore 1
make[2]: uscita dalla directory «/home/callimaco/Documenti/hk_classes/knoda5-hkc-main/utilities»
make[1]: *** [Makefile:445: all-recursive] Errore 1
make[1]: uscita dalla directory «/home/callimaco/Documenti/hk_classes/knoda5-hkc-main»
make: *** [Makefile:377: all] Errore 2
callimaco@deb-portatile:~/Documenti/hk_classes/knoda5-hkc-main$

**Comment by JKBert** [comment]
callimac-o a écrit :
> sorry! is this the error:
> Making all in utilities
> make[2]: ingresso nella directory
> «/home/callimaco/Documenti/hk_classes/knoda5-hkc-main/utilities»
> /bin/bash ../libtool --tag=CXX --mode=link g++ -Wall -W -Wconversion
> -Wshadow -Wcast-qual -Wwrite-strings -I/usr/include/libxml2 -m64 -g -O2
> -o hk_actionquery hk_actionqueryutility.o ../hk_classes/libhk_classes.la
> -ldl
> libtool: link: g++ -Wall -W -Wconversion -Wshadow -Wcast-qual
> -Wwrite-strings -I/usr/include/libxml2 -m64 -g -O2 -o
> .libs/hk_actionquery hk_actionqueryutility.o
> ../hk_classes/.libs/libhk_classes.so -ldl -Wl,-rpath
> -Wl,/usr/lib64/hk_classes
> /usr/bin/ld: hk_actionqueryutility.o: undefined reference to symbol
> 'xmlParseFile@@LIBXML2_2.4.30'
> /usr/bin/ld: /lib/x86_64-linux-gnu/libxml2.so.2: error adding symbols:
> DSO missing from command line
> collect2: error: ld returned 1 exit status
> make[2]: *** [Makefile:514: hk_actionquery] Errore 1
> make[2]: uscita dalla directory
> «/home/callimaco/Documenti/hk_classes/knoda5-hkc-main/utilities»
> make[1]: *** [Makefile:445: all-recursive] Errore 1
> make[1]: uscita dalla directory
> «/home/callimaco/Documenti/hk_classes/knoda5-hkc-main»
> make: *** [Makefile:377: all] Errore 2
> ***@***.***:~/Documenti/hk_classes/knoda5-hkc-main$

	I will check as soon as possible, but if I remember, sqlite ran fine.

**Comment by callimac-o** [comment]
yes, on a pc where i had debian 8 with hk_classes already installed (downloaded from sourceforge by patrik hanak), i upgraded to 11 and re-installed only knoda5 (the one published by you) and sqlite3 works correctly (i use it every day). On the laptop and another computer in the office I have debian9 with knoda4 and hk_classes (sourceforge ph) installed so I can confirm that sqlite3 works.
In the meantime, thank you for your availability
